tests/multi_ptr/multi_ptr_comparison_op.h

```diff
@@ -0,0 +1,591 @@
+  // value_mptr" operator will be called
+  bool nullptr_value_mptr_result_val = true;
+  // Expected value that will be returned after "value_mptr *current operator*
+  // nullptr" operator will be called
+  bool value_mptr_nullptr_result_val = true;
+
+  /**
+   * @brief Verified test result. The object's data, that will call this
+   *        function, will used as expected results
+   * @param retrieved_results Retrieved test results
+   */
+  void verify_results(
+      const mptr_nullptr_mptr_test_results<T> &retrieved_results) const {
+    CHECK(nullptr_nullptr_mptr_result_val ==
+          retrieved_results.nullptr_nullptr_mptr_result_val);
+    CHECK(nullptr_mptr_nullptr_result_val ==
+          retrieved_results.nullptr_mptr_nullptr_result_val);
+    CHECK(nullptr_value_mptr_result_val ==
+          retrieved_results.nullptr_value_mptr_result_val);
+    CHECK(value_mptr_nullptr_result_val ==
+          retrieved_results.value_mptr_nullptr_result_val);
+  }
+};
+
+}  // namespace detail
+
+/**
+ * @brief Provides functor for verification on multi_ptr comparison operators
+ * @tparam T Current data type
+ * @tparam AddrSpaceT sycl::access::address_space enumeration's field
+ * @tparam IsDecoratedT sycl::access::decorated enumeration's field
+ */
+template <typename T, typename AddrSpaceT, typename IsDecoratedT>
+class run_multi_ptr_comparison_op_test {
+  static constexpr sycl::access::address_space space = AddrSpaceT::value;
+  static constexpr sycl::access::decorated decorated = IsDecoratedT::value;
+
+  using multi_ptr_t = sycl::multi_ptr<T, space, decorated>;
+
+  const T m_low_value = 1;
+  const T m_great_value = 2;
+  // Use an array to be sure that we have two elements that has subsequence
+  // memory addresses
+  const T m_values_arr[2] = {m_low_value, m_great_value};
+  sycl::range m_r(1);
+
+  template <typename TestActionT, typename ExpectedTestResultT>
+  void run_test(sycl::queue &queue, TestActionT test_action,
+                const ExpectedTestResultT &expected_results) {
+    // Variable that contains all variables that will be used to verify test
+    // result
+    ExpectedTestResultT test_results;
+    {
+      sycl::buffer<T> array_buffer(m_values_arr, std::size(m_values_arr));
+      sycl::buffer<ExpectedTestResultT> test_result_buffer(&test_results, m_r);
+      queue.submit([&](sycl::handler &cgh) {
+        auto array_acc =
+            array_buffer.template get_access<sycl::access_mode::read>(cgh);
+        auto test_result_acc =
+            test_result_buffer.template get_access<sycl::access_mode::write>(
+                cgh);
+
+        if constexpr (space == sycl::access::address_space::global_space) {
+          cgh.single_task([=] { test_action(array_acc, test_result_acc); });
+        } else {
+          cgh.parallel_for(sycl::nd_range(m_r, m_r), [=](sycl::nd_item item) {
+            test_action(array_acc, test_result_acc);
+          });
+        }
```

Comment:
Why don't we use `parallel_for` for all address spaces?

Reply:
Now all test actions will be launched in `parallel_for`, I've done it [here](https://github.com/KhronosGroup/SYCL-CTS/pull/363/commits/90f72c25c6b6e49ffa2ec5179953f7f3eab373a6#diff-6694f76c9f9ec20ae7ad1201253bd9ce5172455eec7b5a61db5b7b800a5ce57dL151-R152)